Cabal/src/Distribution/Simple/Haddock.hs

```diff
@@ -563,6 +574,11 @@ runHaddock verbosity tmpFileOpts comp platform haddockProg args
 
         notice verbosity $ "Documentation created: " ++ result
 
+        when (fromFlag (argOpen args)) $ do
+          exe <- determineOpenProgramLocation platform
+          case exe of
+            Right open -> runProgramInvocation verbosity (simpleProgramInvocation open [result])
+            Left err -> warn verbosity err
```

Comment:
I think the error should be treated fatally, since here the command is failing to do something the user requested

Reply:
Changed that `warn` to `die’`.